src/plugins/workspace/public/plugin.ts

```diff
@@ -3,16 +3,33 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { Plugin } from '../../../core/public';
+import type { Subscription } from 'rxjs';
+import { Plugin, CoreStart } from '../../../core/public';
 
 export class WorkspacePlugin implements Plugin<{}, {}, {}> {
+  private coreStart?: CoreStart;
+  private currentWorkspaceSubscription?: Subscription;
+  private _changeSavedObjectCurrentWorkspace() {
```

Comment:
what's the benefit of making this as a function? does it affect the lifecycle of corestart object by storing it as a field for the plugin?

Reply:
The benefit is for code readability because we extract all the listener handler codes into a handler function.